src/elfpy/pricing_models/base.py

```diff
@@ -327,6 +328,9 @@ def _calc_spot_price_from_reserves_high_precision(
             market_state.share_reserves
         )
         # p = ((y + s)/(mu*z))^(-tau) = ((2y + cz)/(mu*z))^(-tau)
+        assert market_state.init_share_price != 0, "init_share_price cannot be zero"
```

Comment:
I think it makes more sense to put these in the external function call `calc_spot_price_from_reserves`